src/Workspaces/Core/Portable/FindSymbols/FindLiterals/FindLiteralsSearchEngine.cs

```diff
@@ -0,0 +1,201 @@
+                    _searchKind = SearchKind.NumericLiterals;
+                    break;
+                case decimal d: // unsupported
+                    _searchKind = SearchKind.None;
+                    break;
+                case char c:
+                    _longValue = IntegerUtilities.ToInt64(value);
+                    _searchKind = SearchKind.CharacterLiterals;
+                    break;
+                default:
+                    _longValue = IntegerUtilities.ToInt64(value);
+                    _searchKind = SearchKind.NumericLiterals;
+                    break;
+            }
+        }
+
+        public async Task FindReferencesAsync()
+        {
+            await _progressTracker.AddItemsAsync(1).ConfigureAwait(false);
+            try
+            {
+                if (_searchKind != SearchKind.None)
+                {
+                    await FindReferencesWorkerAsync().ConfigureAwait(false);
+                }
+            }
+            finally
+            {
+                await _progressTracker.ItemCompletedAsync().ConfigureAwait(false);
+            }
+        }
+
+        private async Task FindReferencesWorkerAsync()
+        {
+            var count = _solution.Projects.SelectMany(p => p.DocumentIds).Count();
+            await _progressTracker.AddItemsAsync(count).ConfigureAwait(false);
+
+            foreach (var project in _solution.Projects)
+            {
+                _cancellationToken.ThrowIfCancellationRequested();
+
+                var documentTasks = new List<Task>();
+                foreach (var document in project.Documents)
+                {
+                    documentTasks.Add(ProcessDocumentAsync(document));
+                }
+
+                await Task.WhenAll(documentTasks).ConfigureAwait(false);
+            }
+        }
+
+        private async Task ProcessDocumentAsync(Document document)
+        {
+            try
+            {
+                await ProcessDocumentWorkerAsync(document).ConfigureAwait(false);
+            }
+            finally
+            {
+                await _progressTracker.ItemCompletedAsync().ConfigureAwait(false);
+            }
+        }
+
+        private async Task ProcessDocumentWorkerAsync(Document document)
+        {
+            var index = await SyntaxTreeIndex.GetIndexAsync(
+                document, _cancellationToken).ConfigureAwait(false);
+
+            if (_searchKind == SearchKind.StringLiterals)
+            {
+                if (index.ProbablyContainsStringValue(_stringValue))
+                {
+                    await SearchDocumentAsync(document).ConfigureAwait(false);
+                }
+            }
+            else if (index.ProbablyContainsInt64Value(_longValue))
+            {
+                await SearchDocumentAsync(document).ConfigureAwait(false);
+            }
+        }
+
+        private async Task SearchDocumentAsync(Document document)
+        {
```

Comment:
I'm confused, it looks like ProcessDocumentWorkerAsync only calls this for strings and int64?